File: DSA/Ayush_Mishra_590014844/Day_39_Question_1.c
#include <stdio.h>
#define MAX 100
int stack[MAX], top = -1;

void push(int val) {
    stack[++top] = val;
}

int pop() {
    return stack[top--];
}

void insert_bottom(int val) {
    if (top == -1) {
        push(val);
        return;
    }
    int temp = pop();
    insert_bottom(val);
    push(temp);
}

void reverse() {
    if (top == -1) return;
    int temp = pop();
    reverse();
    insert_bottom(temp);
}

int main() {
    int n, val;
    scanf("%d", &n);
    for (int i = 0; i < n; ++i) {
        scanf("%d", &val);
        push(val);
    }
    reverse();
    for (int i = 0; i <= top; ++i)
        printf("%d ", stack[i]);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_39_Question_1.py
def insert_bottom(stack, val):
    if not stack:
        stack.append(val)
        return
    top = stack.pop()
    insert_bottom(stack, val)
    stack.append(top)

def reverse(stack):
    if not stack:
        return
    top = stack.pop()
    reverse(stack)
    insert_bottom(stack, top)

n = int(input())
stack = list(map(int, input().split()))
reverse(stack)
print(*stack)
